com.unity.ml-agents/Runtime/Sensors/Reflection/ObservableAttribute.cs

```diff
@@ -57,15 +57,16 @@ public class ObservableAttribute : Attribute
         /// <summary>
         /// Supported types and their observation sizes.
         /// </summary>
-        static Dictionary<Type, int> s_TypeSizes = new Dictionary<Type, int>()
+        static Dictionary<Type, (int, Type)> s_TypeToSensorInfo = new Dictionary<Type, (int, Type)>()
         {
-            {typeof(int), 1},
-            {typeof(bool), 1},
-            {typeof(float), 1},
-            {typeof(Vector2), 2},
-            {typeof(Vector3), 3},
-            {typeof(Vector4), 4},
-            {typeof(Quaternion), 4},
+            {typeof(int), (1, typeof(IntReflectionSensor))},
```

Comment:
I can make a struct for these if you think this is ugly. If we ever need a third field, I will; for now, I figure tuples are fine.

Reply:
seems ok for now. Definitely a struct if this grows.